### PR TITLE
S115 aws psoxy improvements

### PR DESCRIPTION
**Pull request by jlorper**
Changes to support end to end run of a connector (tested with GDirectory)
- Implement HEAD calls, as used in validation calls
- Fix linking between Gateway and Lambda functions
- Sanitization rules "caching" as singleton
- URL handling improvements 

### Features
 - [AWS end-to-end connector install](https://app.asana.com/0/1201703347247995/1201732924537315)

### Change implications

 - dependencies added/changed? **no**

### CODE REVIEW

File: infra/modules/aws/main.tf [hunk 1]
```diff
@@ -70,7 +71,12 @@ resource "aws_iam_role" "api-caller" {
           "Effect": "Allow",
           "Action": "execute-api:Invoke",
           "Resource": "arn:aws:execute-api:*:${var.aws_account_id}:*/*/GET/*",
-        }
+        },
+        {
+          "Effect": "Allow",
+          "Action": "execute-api:Invoke",
+          "Resource": "arn:aws:execute-api:*:${var.aws_account_id}:*/*/HEAD/*",
```
Comment (eschultink) [inline infra/modules/aws/main.tf:78]: I don't understand the point in allowing `HEAD` ... seems to complicate things, without adding any practical value

File: infra/modules/aws-psoxy-instance/main.tf [hunk 2]
```diff
@@ -23,25 +44,34 @@ resource "aws_apigatewayv2_integration" "map" {
...
-  function_name = var.function_name
+  function_name = aws_lambda_function.psoxy-instance.function_name
   principal     = "apigateway.amazonaws.com"
 
-  # The /*/*/* part allows invocation from any stage, method and resource path
+
+  # The /*/*/ part allows invocation from any stage, method and resource path
   # within API Gateway REST API.
-  source_arn = "${var.api_gateway.arn}/*/*/*"
+  source_arn = "${var.api_gateway.execution_arn}/*/*/${var.function_name}/{proxy+}"
```
Comment (jlorper) [inline infra/modules/aws-psoxy-instance/main.tf:71]: This was what wasn't connecting the lambda with the gateway propertly

File: infra/modules/aws/main.tf (hunk 1 above)
Comment (jlorper) [reply infra/modules/aws/main.tf:78]: Well, HEAD is slightly faster than GET, but other than that analogous and harmless from the proxy POV.
I could have changed the HEAD calls in Worklytics, but we should adapt proxy to Worklytics, not the other way around.

File: java/core/src/main/java/co/worklytics/psoxy/gateway/impl/CommonRequestHandler.java [hunk 3]
```diff
@@ -37,13 +42,26 @@
     @Inject SanitizerFactory sanitizerFactory;
     @Inject Rules rules;
 
-    private Sanitizer sanitizer;
+    private volatile Sanitizer sanitizer;
+    private final Object $writeLock = new Object[0];
+
+
+    private Sanitizer loadSanitizerRules() {
```
Comment (jlorper) [inline java/core/src/main/java/co/worklytics/psoxy/gateway/impl/CommonRequestHandler.java:49]: load rules as singleton

**Review by eschultink — APPROVED** [review]

**Pull request merged**